app/services/api_authentication_system.rb

```diff
@@ -28,13 +29,15 @@ def permissions(action_name, resource)
   def login_from_api_key
     key = params[:key]
     if key.nil?
-      u = User.find_by_login('guestuser')
+      u = User.find_by_login('guestuserx')
+      if u.nil?
+        @errors = "For key-less access to the API, you must set up the guest user account."
+      end
     else
-      u = User.find_by_apikey(params[:key]) || User.find_by_login('guestuser')
-    end
-
-    if u.nil?
-      @errors = "You must either use a valid API key or set up the guest user account."
+      u = User.find_by_apikey(params[:key])
```

Comment:
nitpicking but can use `key` variable here to prevent an extra lookup.

Reply:
No problem--that was my intention but then I forgot to; and your nitpicking led me to notice and fix a more serious problem.